packages/app-mobile/utils/fs-driver-rn.ts

```diff
@@ -86,12 +88,15 @@ export default class FsDriverRN extends FsDriverBase {
 		for (let i = 0; i < stats.length; i++) {
 			const stat = stats[i];
 			const relativePath = (isScoped ? stat.uri : stat.path).substr(path.length + 1);
-			output.push(this.rnfsStatToStd_(stat, relativePath));
+			const standardStat = this.rnfsStatToStd_(stat, relativePath);
+			output.push(standardStat);
 
 			if (isScoped) {
+				// readUriDirStatsHandleRecursion_ expects stat to have a URI property.
+				// Use the original stat.
 				output = await this.readUriDirStatsHandleRecursion_(stat, output, options);
 			} else {
-				output = await this.readDirStatsHandleRecursion_(path, stat, output, options);
+				output = await this.readDirStatsHandleRecursion_(path, standardStat, output, options);
```

Comment:
`readDirStatsHandleRecursion_` expects a relative and not an absolute path (which is true for standardStat.path, but not stat.path). When the path given to `readDirStatsHandleRecursion_` is absolute, absolute paths are concatenated, resulting in paths like `/storage/emulated/0/tmp//storage/emulated/0/tmp/foo.txt`.